server/services/omnichannel-voip/service.ts

```diff
@@ -17,12 +24,75 @@ export class OmnichannelVoipService extends ServiceClass implements IOmnichannel
 
 	private users: UsersRaw;
 
+	private voipRoom: VoipRoomsRaw;
+
 	constructor(db: Db) {
 		super();
 		this.users = new UsersRaw(db.collection('users'));
+		this.voipRoom = new VoipRoomsRaw(db.collection('rocketchat_room'));
 		this.logger = new Logger('OmnichannelVoipService');
 	}
 
+	private normalizeAgent(agentId: string): ILivechatAgent | { hiddenInfo: true } | undefined {
+		if (!agentId) {
+			return;
+		}
+		if (!settings.get('Livechat_show_agent_info')) {
+			return { hiddenInfo: true };
+		}
+
```

Comment:
@pierre-lehnen-rc and @KevLehman I actually copied this code from app/livechat/server/lib/Helper.js. The hierarchy is as follows :
app/livechat/server/api/v1/room.js -> app/livechat/server/api/lib/livechat.js -> app/livechat/server/lib/Helper.js

I am not sure why this done. Usually the agent reference is used either in subscription (in case of livechat code) or in servedBy field of Room (in case of Voip code).

If Livechat_show_agent_info is false, the code which uses this function (livechat/room or voip/room) assumes that agent exists. It puts just the agentId in the table (Either in subscription or in servedBy of room). Which can cause a trouble. Someone can add an incorrect agentId.
What we are really doing here is using agentId and username while creating a voip room. How would it matter if the Livechat_show_agent_info flag is true or false?
If even showing the agent user name is going against the strategy for having Livechat_show_agent_info, why do we put the username atall? Why dont we just validate that the agentId is valid and just put the agentid in the necessary datastructure? If we need the agent information, fetch it later from the User database.
Let me know your thoughts.

Reply:
The agent username can't be optional on the servedBy attribute because several existing features depend on it (example: Analytics and Queue Manager).

Reply:
Ok. Then shall I remove the check if (!settings.get('Livechat_show_agent_info')) and return the complete agent? We are not using any of the fields other than the id and username. (I am still failing to understand why we are trying to hide the details for room creation)

Reply:
@KevLehman  Aah. Got it. It does not make sense right now as the calls are going to land up from mobile and mobile user has no access to agent name visually. 
I also confirmed with Pierre. I will remove findAgent and normalizeAgent functions for Voip. Will fetch the username and id directly from the user DB in the API file.

Reply:
I have removed the findAgent and normalizeAgent routines. I am now reading the agentName directly from the database in the endpoint file. This would avoid a lot of complications. @pierre-lehnen-rc Please take a look now.